articles/api-auth/dynamic-client-registration.md

```diff
@@ -0,0 +1,205 @@
+
+
+### Update the login page
+
+If you use or would like to use an Auth0 [Hosted Login Page](/hosted-pages/login) with the Dynamic Client feature, you need to use at least version `10.7.x` of Lock, and set `__useTenantInfo: config.isThirdPartyClient` when instantiating Lock.
+
+Sample script:
+
+```html
+<script src="https://cdn.auth0.com/js/lock/10.7/lock.min.js"></script>
+...
+<script>
+  // Decode utf8 characters properly
+  var config = JSON.parse(decodeURIComponent(escape(window.atob('@@config@@'))));
+
+  var connection = config.connection;
+  var prompt = config.prompt;
+  var languageDictionary;
+  var language;
+  if (config.dict && config.dict.signin && config.dict.signin.title) {
+    languageDictionary = { title: config.dict.signin.title };
+  } else if (typeof config.dict === 'string') {
+    language = config.dict;
+  }
+
+  var lock = new Auth0Lock(config.clientID, config.auth0Domain, {
+    auth: {
+      redirectUrl: config.callbackURL,
+      responseType: config.callbackOnLocationHash ? 'token' : 'code',
+      params: config.internalOptions
+    },
+    assetsUrl:  config.assetsUrl,
+    allowedConnections: connection ? [connection] : null,
+    rememberLastLogin: !prompt,
+    language: language,
+    languageDictionary: languageDictionary,
+    closable: false,
+    __useTenantInfo: config.isThirdPartyClient
+  });
+
+  lock.show();
+</script>
+```
+
+## Use dynamic registration
+
+In this section we will see how a third party developer can register and configure a client.
+
+### Register your client
+
+In order to dynamically register a client with Auth0, you need to send an HTTP `POST` message to the [Client Registration endpoint](/api/management/v2#!/oidc/register). Note that Auth0 supports open Dynamic Registration, which means that the [Client Registration endpoint](/api/management/v2#!/oidc/register) will accept a registration request without an [OAuth 2.0 Access Tokens](/tokens/access-token).
+
+To create a client with the name `My Example` and the callback URLs `https://client.example.com/callback` and `https://client.example.com/callback2`, you would use the following.
+
+```har
+{
+  "method": "POST",
+  "url": "https://${account.namespace}/oidc/register",
+  "headers": [
+    { "name": "Content-Type", "value": "application/json" }
+  ],
+  "postData": {
+    "mimeType": "application/json",
+    "text": "{\"client_name\":\"My Dynamic Client\",\"redirect_uris\": [\"https://client.example.com/callback\", \"https://client.example.com/callback2\"]}"
+  }
+}
+```
+
+Where:
+- `client_name`: The name of the Dynamic Client to be created.
+- `redirect_uris`: An array of URLs that Auth0 will deem valid to call at the end of an Authentication flow.
+
+Optionally, you can set a value for `token_endpoint_auth_method`, which can be `none` or `client_secret_post` (default value).
+
+**NOTE**: You can set several attributes. For a complete listing refer to the [/oidc/register endpoint](/api/management/v2#!/oidc/register).
```

Comment:
Remove this. Also, the only attributes we allow in the payload are,
- client_name (required)
- redirect_urs (required)
- token_endpoint_auth_method (optional) (default: client_secret_post)